src/plugins/mf_priority.cpp

```diff
@@ -774,6 +776,31 @@ static int priority_cb (flux_plugin_t *p,
                                          &bank_it->second,
                                          NULL) < 0)
                 flux_log_error (h, "flux_jobtap_job_aux_set");
+
+            // if using a default project, add it to jobspec
+            if (project == NULL) {
+                project = const_cast <char *>
+                    (bank_it->second.def_project.c_str ());
+                if (update_jobspec_project (p, userid, bank, project) < 0) {
+                    flux_jobtap_raise_exception (p, FLUX_JOBTAP_CURRENT_JOB,
+                                                "mf_priority", 0,
+                                                "failed to update jobspec "
+                                                " with project name");
+
+                    return -1;
+                }
+            }
+        }
+    }
+
+    // if using a default project, add it to jobspec
+    if (project == NULL) {
+        if (update_jobspec_project (p, userid, bank, project) < 0) {
+            flux_jobtap_raise_exception (p, FLUX_JOBTAP_CURRENT_JOB,
+                                         "mf_priority", 0, "failed to update "
+                                         "jobspec with project name");
```

Comment:
is there any way to combine this duplicated code? I might need help understanding why `update_jobspec_project()` needs to be called two different places like this here, sorry!

Reply:
Ah, you're so right! Thanks for pointing this out. I don't know why I have this code duplicated; it doesn't need to be. I think I must have thought that if a job is submitted and held in PRIORITY because a user is waiting for their information to be added to the plugin, it required a special case in `priority_cb ()` to add the project name, but that's not the case. Sorry for the confusion!

I'll remove this duplicate code and add a test which shows that the project name gets updated in the case where a user's job is held in PRIORITY as the plugin waits for (and then receives) information from the flux-accounting DB.